File: c-sharp/ApiTests/PremDbHandlerTest.cs
using FluentAssertions;
using Npb.Agview.Api.Example;
using Xunit;

namespace Npb.Agview.ApiTests.Example
{

    public class PremDbHandlerTest
    {
        private readonly PremDbHandler _sut;
        private readonly string PremsFilePath = Constants.PremsFilePath;
        private readonly string PremAddressesFilePath = Constants.PremAddressesFilePath;

        private const string UsdaPin1 = "1277XH9";
        private const string UsdaPin2 = "0615PKA";
        private const string PremName1 = "David's Prem";
        private const string PremName2 = "Nate's Prem";
        private const string IceContactEmail = "ice@vanila.com";
        private const string IceContactPhone = "12165551234";
        private const string LocationType = "Nursery";
        private const string Species = "Swine";
        private const int SiteCapacityNumberAnimals = 1000;
        private const int SiteCapacityNumberBarns = 3;
        private const int NumberOfAnimalsOnSite = 20;
        private const string StreetAddress1 = "951 S. Green Rd";
        private const string City1 = "South Euclid";
        private const string Zip1 = "44121";
        private const string State1 = "OH";
        private const string StreetAddress2 = "555 Superior Ave";
        private const string City2 = "Cleveland";
        private const string Zip2 = "44108";
        private const string State2 = "OH";

        public PremDbHandlerTest()
        {
            _sut = new PremDbHandler(PremsFilePath, PremAddressesFilePath);
        }

        [Fact]
        public void GetsPrems()
        {
            var expected1 = new Prem
            {
              UsdaPin = UsdaPin1,
              PremName = PremName1,
              SiteCapacityNumberAnimals = SiteCapacityNumberAnimals,
              IceContactEmail = IceContactEmail,
              IceContactPhone = IceContactPhone,
              LocationType = LocationType,
              Species = Species,
              SiteCapacityNumberBarns = SiteCapacityNumberBarns,
              NumberOfAnimalsOnSite = NumberOfAnimalsOnSite
            };

            var expected2 = new Prem
            {
                UsdaPin = UsdaPin2,
                PremName = PremName2,
                SiteCapacityNumberAnimals = SiteCapacityNumberAnimals,
                IceContactEmail = IceContactEmail,
                IceContactPhone = IceContactPhone,
                LocationType = LocationType,
                Species = Species,
                SiteCapacityNumberBarns = SiteCapacityNumberBarns,
                NumberOfAnimalsOnSite = NumberOfAnimalsOnSite
            };

            var actual = _sut.GetPremsToLoad();

            actual.Should().HaveCount(2);
            actual.Should().ContainEquivalentOf(expected1);
            actual.Should().ContainEquivalentOf(expected2);
        }

        [Fact]
        public void GetsPremColumnNames()
        {
            var actual = _sut.GetPremColumnNames();

            string.Join(",", actual).Should().Be("usda_pin,prem_name,site_capacity_number_animals,ice_contact_email,ice_contact_phone,location_type,species,site_capacity_number_barns,number_of_animals_on_site");
        }

        [Fact]
        public void GetsPremAddresses()
        {
            var expected1 = new PremAddress
            {
                UsdaPin = UsdaPin1,
                StreetAddress = StreetAddress1,
                City = City1,
                Zip = Zip1,
                State = State1
            };
            var expected2 = new PremAddress
            {
                UsdaPin = UsdaPin2,
                StreetAddress = StreetAddress2,
                City = City2,
                Zip = Zip2,
                State = State2
            };

            var actual = _sut.GetPremAddressesToLoad();

            actual.Should().HaveCount(2);
            actual.Should().ContainEquivalentOf(expected1);
            actual.Should().ContainEquivalentOf(expected2);
        }

        [Fact]
        public void GetsPremAddressColumnNames()
        {
            var actual = _sut.GetPremAddressColumnNames();

            string.Join(",", actual).Should().Be("usda_pin,street_address,city,state,zip");
        }
    }

}

File: java/src/test/java/org/npb/agview/api/example/PremDbHandlerTest.java
package org.npb.agview.api.example;

import org.hamcrest.Matchers;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.*;
import static org.hamcrest.Matchers.*;
import static org.npb.agview.api.example.Constants.*;

class PremDbHandlerTest {

    private PremDbHandler sut;

    private static final String USDA_PIN1 = "1277XH9";
    private static final String USDA_PIN2 = "0615PKA";
    private static final String PREM_NAME1 = "David's Prem";
    private static final String PREM_NAME2 = "Nate's Prem";
    private static final String SPECIES = "Swine";
    private static final String ICE_CONTACT_PHONE = "12165551234";
    private static final String ICE_CONTACT_EMAIL = "ice@vanila.com";
    private static final String LOCATION_TYPE = "Nursery";
    private static final int SITE_CAPACITY_NUMBER_BARNS = 3;
    private static final int SITE_CAPACITY_NUMBER_ANIMALS = 1000;
    private static final int NUMBER_OF_ANIMALS_ONSITE = 20;
    private static final String STREET_ADDRESS1 = "951 S. Green Rd";
    private static final String CITY1 = "South Euclid";
    private static final String STATE1 = "OH";
    private static final String ZIP1 = "44121";
    private static final String STREET_ADDRESS2 = "555 Superior Ave";
    private static final String CITY2 = "Cleveland";
    private static final String STATE2 = "OH";
    private static final String ZIP2 = "44108";

    @BeforeEach
    public void setup() {
        sut = new PremDbHandler(DB_DIRECTORY+"/prem.csv",
                DB_DIRECTORY+"/prem_address.csv", new DbHandler());
    }

    @Test
    public void getsPrems() {
        var expectedPrem1 = new Prem(USDA_PIN1, PREM_NAME1, SPECIES, ICE_CONTACT_PHONE, ICE_CONTACT_EMAIL, LOCATION_TYPE,
                SITE_CAPACITY_NUMBER_BARNS, SITE_CAPACITY_NUMBER_ANIMALS, NUMBER_OF_ANIMALS_ONSITE);
        var expectedPrem2 = new Prem(USDA_PIN2, PREM_NAME2, SPECIES, ICE_CONTACT_PHONE, ICE_CONTACT_EMAIL, LOCATION_TYPE,
                SITE_CAPACITY_NUMBER_BARNS, SITE_CAPACITY_NUMBER_ANIMALS, NUMBER_OF_ANIMALS_ONSITE);


        var actual = sut.getPremsToLoad();

        assertThat(actual, Matchers.containsInAnyOrder(expectedPrem1, expectedPrem2));
    }

    @Test
    public void GetsPremColumnNames() {
        var actual = sut.getPremColumnNames();

        assertThat(actual, contains("usda_pin","prem_name","site_capacity_number_animals","ice_contact_email","ice_contact_phone","location_type","species","site_capacity_number_barns","number_of_animals_on_site"));
    }

    @Test
    public void getsPremAddresses() {
        var expectedAddress1 = new PremAddress(USDA_PIN1, STREET_ADDRESS1, CITY1, STATE1, ZIP1);
        var expectedAddress2 = new PremAddress(USDA_PIN2, STREET_ADDRESS2, CITY2, STATE2, ZIP2);

        var actual = sut.getPremAddressesToLoad();

        assertThat(actual, containsInAnyOrder(expectedAddress1, expectedAddress2));
    }

    @Test
    public void GetsPremAddressColumnNames() {
        var actual = sut.getPremAddressesColumnNames();

        assertThat(actual, contains("usda_pin","street_address","city","state","zip"));
    }

}
